spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/RabbitUtils.java

```diff
@@ -324,21 +324,11 @@ public static boolean isExchangeDeclarationFailure(Exception e) {
 	 * @param logger the logger to use for debug.
 	 * @return true to requeue.
 	 * @since 2.0
+	 * @deprecated in favor of {@code ContainerUtils#shouldRequeue(boolean, Throwable, Log)}.
 	 */
+	@Deprecated
 	public static boolean shouldRequeue(boolean defaultRequeueRejected, Throwable throwable, Log logger) {
-		boolean shouldRequeue = defaultRequeueRejected ||
-				throwable instanceof MessageRejectedWhileStoppingException;
-		Throwable t = throwable;
-		while (shouldRequeue && t != null) {
-			if (t instanceof AmqpRejectAndDontRequeueException) {
-				shouldRequeue = false;
-			}
-			t = t.getCause();
-		}
-		if (logger.isDebugEnabled()) {
-			logger.debug("Rejecting messages (requeue=" + shouldRequeue + ")");
-		}
-		return shouldRequeue;
+		throw new UnsupportedOperationException("Use ContainerUtils");
```

Comment:
Why can't we delegate here to the mentioned `ontainerUtils.shouldRequeue()` then?
This seems for me as breaking change now...

I think even if we need a reflection here, it still would be better to keep functionality unbroken.

Although we are moving stuff around anyway to fix tangles... :smile: